x-pack/legacy/plugins/apm/public/components/shared/TransactionActionMenu/sections.ts

```diff
@@ -0,0 +1,303 @@
+        }
+      ),
+      href: getDiscoverHref({
+        basePath,
+        query: getDiscoverQuery(transaction),
+        location
+      }),
+      condition: true
+    }
+  ];
+
+  const sectionRecord: SectionRecord = {
+    observability: [
+      {
+        key: 'podDetails',
+        title: i18n.translate('xpack.apm.transactionActionMenu.pod.title', {
+          defaultMessage: 'Pod details'
+        }),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.pod.subtitle',
+          {
+            defaultMessage:
+              'View logs and metrics of this pod to get further details.'
+          }
+        ),
+        actions: podActions
+      },
+      {
+        key: 'containerDetails',
+        title: i18n.translate(
+          'xpack.apm.transactionActionMenu.container.title',
+          {
+            defaultMessage: 'Container details'
+          }
+        ),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.container.subtitle',
+          {
+            defaultMessage:
+              'View logs and metrics of this container to get further details.'
+          }
+        ),
+        actions: containerActions
+      },
+      {
+        key: 'hostDetails',
+        title: i18n.translate('xpack.apm.transactionActionMenu.host.title', {
+          defaultMessage: 'Host details'
+        }),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.host.subtitle',
+          {
+            defaultMessage: 'View host logs to get further details.'
+          }
+        ),
+        actions: hostActions
+      },
+      {
+        key: 'traceDetails',
+        title: i18n.translate('xpack.apm.transactionActionMenu.trace.title', {
+          defaultMessage: 'Trace details'
+        }),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.trace.subtitle',
+          {
+            defaultMessage: 'View the trace logs to get further details.'
+          }
+        ),
+        actions: logActions
+      },
+      {
+        key: 'statusDetails',
+        title: i18n.translate('xpack.apm.transactionActionMenu.status.title', {
+          defaultMessage: 'Status details'
+        }),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.status.subtitle',
+          {
+            defaultMessage: 'View the status to get further details.'
+          }
+        ),
+        actions: uptimeActions
+      }
+    ],
+    kibana: [{ key: 'kibana', actions: kibanaActions }]
+  };
+
+  // Filter out actions that shouldnt be shown and sections without any actions.
+  return Object.values(sectionRecord).map(sections =>
+    sections
+      .map(sectionItem => {
+        const actions = sectionItem.actions.filter(
+          (action: Action) => action.condition
+        );
+        return { ...sectionItem, actions };
+      })
+      .filter(sectionItem => sectionItem.actions.length > 0)
+  );
```

Comment:
I think the naming is still off here. `sections` => `sectionItems` OR `section`. `sections` is the result of `Object.values(sectionRecord)`. If you change that, you can also see that we're not filtering out sections that do not have any items (I think).

Reply:
I renamed it to SectionRecord ->  Sections -> Section -> Actions. Also created some unit test to make sure the filters are right.